pkg/plugins/policies/meshaccesslog/plugin/xds/configurer.go

```diff
@@ -0,0 +1,147 @@
+	envoy_hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	envoy_tcp "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/tcp_proxy/v3"
+	"github.com/pkg/errors"
+
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	accesslog "github.com/kumahq/kuma/pkg/envoy/accesslog/v3"
+	api "github.com/kumahq/kuma/pkg/plugins/policies/meshaccesslog/api/v1alpha1"
+	util_proto "github.com/kumahq/kuma/pkg/util/proto"
+	"github.com/kumahq/kuma/pkg/xds/envoy"
+	listeners_v3 "github.com/kumahq/kuma/pkg/xds/envoy/listeners/v3"
+)
+
+const (
+	defaultNetworkAccessLogFormat = `[%START_TIME%] %RESPONSE_FLAGS% %KUMA_MESH% %KUMA_SOURCE_ADDRESS_WITHOUT_PORT%(%KUMA_SOURCE_SERVICE%)->%UPSTREAM_HOST%(%KUMA_DESTINATION_SERVICE%) took %DURATION%ms, sent %BYTES_SENT% bytes, received: %BYTES_RECEIVED% bytes`
+	defaultHttpAccessLogFormat    = `[%START_TIME%] %KUMA_MESH% "%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%" %RESPONSE_CODE% %RESPONSE_FLAGS% %BYTES_RECEIVED% %BYTES_SENT% %DURATION% %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)% "%REQ(X-FORWARDED-FOR)%" "%REQ(USER-AGENT)%" "%REQ(X-B3-TRACEID?X-DATADOG-TRACEID)%" "%REQ(X-REQUEST-ID)%" "%REQ(:AUTHORITY)%" "%KUMA_SOURCE_SERVICE%" "%KUMA_DESTINATION_SERVICE%" "%KUMA_SOURCE_ADDRESS_WITHOUT_PORT%" "%UPSTREAM_HOST%"`
+)
+
+type Configurer struct {
+	Mesh               string
+	TrafficDirection   envoy.TrafficDirection
+	SourceService      string
+	DestinationService string
+	Backend            *api.MeshAccessLog_Backend
+	Dataplane          *core_mesh.DataplaneResource
+}
+
+func (c *Configurer) handlePlain(formatString string) (*accesslog.AccessLogFormat, error) {
+	envoyFormat, err := accesslog.ParseFormat(formatString + "\n")
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "invalid access log format string: %s", formatString)
+	}
+
+	variables := accesslog.InterpolationVariables{
+		accesslog.CMD_KUMA_SOURCE_ADDRESS:              net.JoinHostPort(c.Dataplane.GetIP(), "0"), // deprecated variable
+		accesslog.CMD_KUMA_SOURCE_ADDRESS_WITHOUT_PORT: c.Dataplane.GetIP(),                        // replacement variable
+		accesslog.CMD_KUMA_SOURCE_SERVICE:              c.SourceService,
+		accesslog.CMD_KUMA_DESTINATION_SERVICE:         c.DestinationService,
+		accesslog.CMD_KUMA_MESH:                        c.Mesh,
+		accesslog.CMD_KUMA_TRAFFIC_DIRECTION:           string(c.TrafficDirection),
+	}
+
+	envoyFormat, err = envoyFormat.Interpolate(variables)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to interpolate access log format string with Kuma-specific variables: %s", formatString)
+	}
+
+	return envoyFormat, nil
+}
+
+func (c *Configurer) envoyAccessLog(defaultFormat string) (*envoy_accesslog.AccessLog, error) {
+	var format *api.MeshAccessLog_Format
+	if f := c.Backend.GetFile().GetFormat(); f != nil {
+		format = f
+	} else if f := c.Backend.GetTcp().GetFormat(); f != nil {
+		format = f
+	}
+
+	// TODO json
```

Comment:
should we create an issue for that?

Reply:
I think I'd just leave the original issue open